### ISSUE TITLE
NVDA crashes MS Word when you print complex document via PDF creator

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#3102]
**Reported by ondrosik on 2013-03-23 19:22**
First, I realize, that this issue is really complicated to reproduce, but I will try my best.
I use PDF creator 1.6.0, Word 2003 and NVDA branch 2013.1 changeset 5975 with addons Resource monitor, dropbox, extended winamp, Instant translate and OCR.
.
I had no problems till today, when I tryed to convert 42 pages document with contents, headings and footnotes. Everithing also works, when i type something to empty document and save it as PDF.
Steps to reproduce:
1) open the document and click print >pdf creator>OK.
2) Word will crash with standard dialog "send report, don't send" buttons. Click don't send, and close window with word, because is still open and program stops responding.
3) close the PDF creator window. (I tryed to save the offered PDF, but there were first three pages only).

Now reproduce the same, but
1) open the document and click print >pdf creator> find the OK button.
2) Quit NVDA and wait for 5 seconds.
3) press space to activate the OK button.
4) Wait for a while
5) Start NVDA.
here is the point. You should be dyrectly in the pdf creator and there are no Word error dialogs. So when I saved the PDF, i got full 42 pages.

As I said, when I try to do this with small document, everithing works also with NVDA enabled, this is probably first time when I need convert something more complex to PDF.

Here is the log, I know, that NVDA is not crashing itself, but there is bunch of errors, so you might be interested:
Here is the point, where I pressed space bar in OK button.
IO - speech.speak (19:53:23):
Speaking [\- Microsoft Word'](u'rosik_profi_Radia_proglas_23_03_13.doc)
IO - speech.speak (19:53:23):
Speaking [- speech.speak (19:53:23):
Speaking [u'Dokument programu Microsoft Word  edita\u010dn\xe9  viacriadkov\xe9'](u'rosik_profi_Radia_proglas_23_03_13.doc']
IO) (MS word document, multyline edit)
IO - speech.speak (19:53:23):
Speaking [8', u'KATOL\xcdCKA UNIVERZITA V RU\u017dOMBERKU\n'](u'Strana) (page 8, first line of the page "atholic University...")
DEBUGWARNING - watchdog._watcher (19:53:26):
Trying to recover from freeze, core stack:
  File "nvda.pyw", line 157, in <module>
  File "core.pyc", line 307, in main
  File "wx_core.pyc", line 8010, in MainLoop
  File "wx_core.pyc", line 7306, in MainLoop
  File "core.pyc", line 284, in Notify
  File "queueHandler.pyc", line 76, in pumpAll
  File "queueHandler.pyc", line 47, in flushQueue
  File "eventHandler.pyc", line 54, in _queueEventCallback
  File "eventHandler.pyc", line 136, in executeEvent
  File "eventHandler.pyc", line 84, in __init__
  File "eventHandler.pyc", line 91, in next
  File "NVDAObjects__init__.pyc", line 823, in event_gainFocus
  File "NVDAObjects__init__.pyc", line 767, in reportFocus
  File "comtypesMonkeyPatches.pyc", line 51, in newCpbDel
  File "comtypes__init__.pyc", line 841, in **del**
  File "comtypes__init__.pyc", line 1092, in Release

DEBUGWARNING - watchdog._watcher (19:53:26):
Trying to recover from freeze, core stack:
  File "nvda.pyw", line 157, in <module>
  File "core.pyc", line 307, in main
  File "wx_core.pyc", line 8010, in MainLoop
  File "wx_core.pyc", line 7306, in MainLoop
  File "core.pyc", line 284, in Notify
  File "queueHandler.pyc", line 76, in pumpAll
  File "queueHandler.pyc", line 44, in flushQueue
  File "comtypesMonkeyPatches.pyc", line 51, in newCpbDel
  File "comtypes__init__.pyc", line 841, in **del**
  File "comtypes__init__.pyc", line 1092, in Release

DEBUGWARNING - watchdog._watcher (19:53:27):
Trying to recover from freeze, core stack:
  File "nvda.pyw", line 157, in <module>
  File "core.pyc", line 307, in main
  File "wx_core.pyc", line 8010, in MainLoop
  File "wx_core.pyc", line 7306, in MainLoop
  File "core.pyc", line 284, in Notify
  File "queueHandler.pyc", line 76, in pumpAll
  File "queueHandler.pyc", line 44, in flushQueue
  File "comtypesMonkeyPatches.pyc", line 51, in newCpbDel
  File "comtypes__init__.pyc", line 841, in **del**
  File "comtypes__init__.pyc", line 1092, in Release

DEBUGWARNING - watchdog._watcher (19:53:28):
Trying to recover from freeze, core stack:
  File "nvda.pyw", line 157, in <module>
  File "core.pyc", line 307, in main
  File "wx_core.pyc", line 8010, in MainLoop
  File "wx_core.pyc", line 7306, in MainLoop
  File "core.pyc", line 284, in Notify
  File "queueHandler.pyc", line 76, in pumpAll
  File "queueHandler.pyc", line 44, in flushQueue
  File "comtypesMonkeyPatches.pyc", line 51, in newCpbDel
  File "comtypes__init__.pyc", line 841, in **del**
  File "comtypes__init__.pyc", line 1092, in Release

DEBUGWARNING - watchdog._watcher (19:53:28):
Trying to recover from freeze, core stack:
  File "nvda.pyw", line 157, in <module>
  File "core.pyc", line 307, in main
  File "wx_core.pyc", line 8010, in MainLoop
  File "wx_core.pyc", line 7306, in MainLoop
  File "core.pyc", line 284, in Notify
  File "queueHandler.pyc", line 76, in pumpAll
  File "queueHandler.pyc", line 44, in flushQueue
  File "comtypesMonkeyPatches.pyc", line 51, in newCpbDel
  File "comtypes__init__.pyc", line 841, in **del**
  File "comtypes__init__.pyc", line 1092, in Release

DEBUGWARNING - watchdog._watcher (19:53:29):
Trying to recover from freeze, core stack:
  File "nvda.pyw", line 157, in <module>
  File "core.pyc", line 307, in main
  File "wx_core.pyc", line 8010, in MainLoop
  File "wx_core.pyc", line 7306, in MainLoop
  File "core.pyc", line 284, in Notify
  File "queueHandler.pyc", line 76, in pumpAll
  File "queueHandler.pyc", line 44, in flushQueue
  File "comtypesMonkeyPatches.pyc", line 51, in newCpbDel
  File "comtypes__init__.pyc", line 841, in **del**
  File "comtypes__init__.pyc", line 1092, in Release

DEBUGWARNING - watchdog._watcher (19:53:29):
Trying to recover from freeze, core stack:
  File "nvda.pyw", line 157, in <module>
  File "core.pyc", line 307, in main
  File "wx_core.pyc", line 8010, in MainLoop
  File "wx_core.pyc", line 7306, in MainLoop
  File "core.pyc", line 284, in Notify
  File "queueHandler.pyc", line 76, in pumpAll
  File "queueHandler.pyc", line 44, in flushQueue
  File "comtypesMonkeyPatches.pyc", line 51, in newCpbDel
  File "comtypes__init__.pyc", line 841, in **del**
  File "comtypes__init__.pyc", line 1092, in Release

DEBUGWARNING - watchdog._watcher (19:53:30):
Trying to recover from freeze, core stack:
  File "nvda.pyw", line 157, in <module>
  File "core.pyc", line 307, in main
  File "wx_core.pyc", line 8010, in MainLoop
  File "wx_core.pyc", line 7306, in MainLoop
  File "core.pyc", line 283, in Notify
  File "IAccessibleHandler.pyc", line 843, in pumpAll
  File "IAccessibleHandler.pyc", line 588, in processGenericWinEvent
  File "IAccessibleHandler.pyc", line 507, in winEventToNVDAEvent
  File "NVDAObjects\IAccessible__init__.pyc", line 37, in getNVDAObjectFromEvent
  File "IAccessibleHandler.pyc", line 333, in accessibleObjectFromEvent
  File "oleacc.pyc", line 240, in AccessibleObjectFromEvent

DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (19:53:30):
oleacc.AccessibleObjectFromEvent with window 983534, objectID -4 and childID 0: 
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (19:53:31):
oleacc.AccessibleObjectFromEvent with window 3735870, objectID -4 and childID 0: 
DEBUGWARNING - watchdog._watcher (19:53:32):
Trying to recover from freeze, core stack:
  File "nvda.pyw", line 157, in <module>
  File "core.pyc", line 307, in main
  File "wx_core.pyc", line 8010, in MainLoop
  File "wx_core.pyc", line 7306, in MainLoop
  File "core.pyc", line 284, in Notify
  File "queueHandler.pyc", line 76, in pumpAll
  File "queueHandler.pyc", line 47, in flushQueue
  File "eventHandler.pyc", line 54, in _queueEventCallback
  File "eventHandler.pyc", line 136, in executeEvent
  File "eventHandler.pyc", line 84, in __init__
  File "eventHandler.pyc", line 91, in next
  File "NVDAObjects\IAccessible__init__.pyc", line 1281, in event_caret
  File "NVDAObjects\window\winword.pyc", line 355, in event_caret
  File "NVDAObjects__init__.pyc", line 893, in makeTextInfo
  File "NVDAObjects\window\winword.pyc", line 155, in **init**
  File "comtypesMonkeyPatches.pyc", line 32, in new__getattr__
  File "comtypes\client\lazybind.pyc", line 149, in **getattr**
  File "comtypes\automation.pyc", line 664, in _invoke

DEBUGWARNING - eventHandler.executeEvent (19:53:32):
error executing event: caret on <NVDAObjects.Dynamic_IAccessibleWordDocumentWindowNVDAObject object at 0x03EF9E30> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyc", line 136, in executeEvent
  File "eventHandler.pyc", line 84, in **init**
  File "eventHandler.pyc", line 91, in next
  File "NVDAObjects\IAccessible__init__.pyc", line 1281, in event_caret
  File "NVDAObjects\window\winword.pyc", line 355, in event_caret
  File "NVDAObjects__init__.pyc", line 893, in makeTextInfo
  File "NVDAObjects\window\winword.pyc", line 155, in **init**
  File "watchdog.pyc", line 147, in _COMError_init

I don't attached the document, as it is personal stuff and I am not sure if not registered users should download the attachments from trac, but i can send it privately via e-mail.

what will I do next:
Remove all addons, just to minimize potential errors from there (but I don't expect, that this should help as mentioned addons are from the addons official site)
- Try it with another document, for example take some long book, place there footnotes and stuff and see, if this should be reproduced with other documents.

**Comment by bhavyashah** [comment]
The original author of this ticket noted in https://github.com/nvaccess/nvda/issues/3102#issuecomment-155305881 that the reported crash was specific to Windows XP and/or Word 2003, both of which are too old software to address anymore. Thus, I suggest closing. @ehollig

**Comment by ehollig** [comment]
Closing, as this is a Word 2003 and XP specific issue. Note that I did try this using the attached document and Word 2016 and was able to print to PDF without errors. It also looks like the PDF program that was used has been updated to PDF creator 2.5.1. If this is still an issue using the latest version of this program, we can reopen. However, it sounds like from https://github.com/nvaccess/nvda/issues/3102#issuecomment-155305882 that this issue is specific to XP.
